src/internal/modules/invoice-accounts/controllers/invoice-accounts.js

```diff
@@ -53,35 +55,74 @@ const postCompany = async (request, h) => {
 
 const getAddress = async (request, h) => {
   const { regionId, companyId } = request.params;
-  const addresses = await dataService.getCompanyAddresses(companyId);
   // get the session data to check if the address has been set and if it is new or existing
   const session = dataService.sessionManager(request, regionId, companyId);
+  const addresses = await helpers.getAllAddresses(companyId, session);
```

Comment:
this helper method gets all the addresses for the company and the agent company. If the user selected the agent company they could potentially select an address from the original company and I am not sure we would want this to happen. In the database the address could then be linked to the original company and the agent company. 

I think it would be better to only load the addresses for the selected company into this form

Reply:
Good point.

I ran this past Chris and Ruth.

The outcome was: It's OK to show both companies' addresses because it avoids duplications in data entry to the database and expedites the user journey. 

We discussed that we can change this later down the line.

(So I will leave it as it is)